### PR TITLE
Get hmftools working with java 11

### PR DESCRIPTION
**Pull request by hongwingl**
@nedleitch and I made some effort to get hmftools working with java11. This means a few of the dependencies had to be upgraded, and some very small, non functional code changes to come with it. They are:

- Upgrade jooq 3.9.4 -> 3.15.4. The API is slightly changed, any date insert uses `java.time.LocalDate` instead of `java.sql.Date`
- Replace jmockit with mockito. jmockit has some issues running in java 11. This is only used in purple testing.
- Upgrade kotlin 1.2.61 -> 1.5.32.
- Replaced kotlintest with kotest. I could not get kotlintest working in java 11, and it appears kotlintest is an abandoned fork of kotest.

In this change the pom.xml has been modified to compile for Java 11.

### CODE REVIEW

**Review by nedleitch — COMMENTED** [review]
Nice job picking this up!

i should note that the branch fails to build for me under Linux running with OpenJDK 11.0.12. I am able to build from master with JDK 8. Won't go into details here but if you want to get in touch on Slack we can review.

**Review by pauldwolfe — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -281,7 +280,7 @@
             <dependency>
                 <groupId>org.apache.logging.log4j</groupId>
                 <artifactId>log4j-slf4j-impl</artifactId>
-                <version>${apache.log4j.slf4j.version}</version>
+                <version>${apache.log4j.version}</version>
```
[the remark this answers is not in the document]
Comment (hongwingl) [reply pom.xml:283]: Not really. I just thought might as well use the same version.

**Comment by nedleitch** [comment]
I've added some more changes after trying to get the build to work on my machine. Some of these are a continuation of what Hong started with removing the `Util.sqlDate` calls. 

The other changes were to switch to Mockito for Purple as it was failing to initialise JMockit. I updated to the latest JMockit version which is quite old by now already and it didn't resolve the problems so I just switched since these tests don't use advanced JMockit features. Suggest we investigate switching project-wide to another framework as JMockit development activity looks to be pretty low.

In both cases I'm concerned about real-world impacts. My concern was just getting it to build but feels like if these changes are accepted they should have a thorough functional review.

**Comment by hongwingl** [comment]
I pushed another change to upgrade kotlin so it can generate java 11 bytecode.

**Pull request merged**